sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveQl.scala

```diff
@@ -386,6 +386,13 @@ private[hive] object HiveQl {
     (db, tableName)
   }
 
+  protected def extractTableIdent(tableNameParts: Node): IndexedSeq[String] = {
+    tableNameParts.getChildren.map { case Token(part, Nil) => cleanIdentifier(part) } match {
+      case Seq(tableOnly) => IndexedSeq(tableOnly)
+      case Seq(databaseName, table) => IndexedSeq(table, databaseName)
```

Comment:
I'd add a default case here that does something like `case other => sys.error(s"Hive only supports tables names like 'tableName' or 'databaseName.tableName', found '$other'")`